src/fd_table.c

```diff
@@ -114,14 +116,15 @@ static uvwasi_errno_t uvwasi__get_type_and_rights(uv_file fd,
   if (r != 0)
     return uvwasi__translate_uv_error(r);
 
-  if (S_ISREG(mode)) {
+  if ((mode & S_IFMT) == S_IFREG) {
     *type = UVWASI_FILETYPE_REGULAR_FILE;
     *rights_base = UVWASI__RIGHTS_REGULAR_FILE_BASE;
     *rights_inheriting = UVWASI__RIGHTS_REGULAR_FILE_INHERITING;
-  } else if (S_ISDIR(mode)) {
+  } else if ((mode & S_IFMT) == S_IFDIR) {
     *type = UVWASI_FILETYPE_DIRECTORY;
     *rights_base = UVWASI__RIGHTS_DIRECTORY_BASE;
     *rights_inheriting = UVWASI__RIGHTS_DIRECTORY_INHERITING;
+#ifdef S_ISSOCK
```

Comment:
Does `S_IFSOCK` exist on Windows?

Reply:
No, because Windows does not treat sockets as files.